soul-admin/src/main/java/org/dromara/soul/admin/service/impl/UpstreamCheckService.java

```diff
@@ -168,21 +182,25 @@ private void check(final String selectorName, final List<DivideUpstream> upstrea
     }
 
     private void updateSelectorHandler(final String selectorName, final List<DivideUpstream> upstreams) {
-        SelectorDO selector = selectorService.findByName(selectorName);
-        if (Objects.nonNull(selector)) {
-            SelectorData selectorData = selectorService.buildByName(selectorName);
-            if (upstreams == null) {
-                selector.setHandle("");
-                selectorData.setHandle("");
-            } else {
-                String handler = GsonUtils.getInstance().toJson(upstreams);
-                selector.setHandle(handler);
+        SelectorDO selectorDO = selectorMapper.selectByName(selectorName);
+        if (Objects.nonNull(selectorDO)) {
+            String handler = CollectionUtils.isEmpty(upstreams) ? "" : GsonUtils.getInstance().toJson(upstreams);
+            List<ConditionData> conditionDataList = selectorConditionMapper
```

Comment:
what is mean？

Reply:
1. Remove the dependency of the Class UpstreamCheckService on the Class SelectorService and modify it to rely on the Class Mapper, because the Class SelectorService will depend on the Class UpstreamCheckService.
2. The value of the selector is already known when the selectorData is generated, reducing one query operation.